### PR TITLE
feat: Provides 'Adding a Visual Studio Code Extension' module

### PR DESCRIPTION
**Pull request by benoitf**
## What does this pull request change
Introduce `Adding a Visual Studio Code Extension` module

## What issues does this pull request fix or reference
https://github.com/eclipse/che/issues/21296

## Specify the version of the product this pull request applies to
main

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [x] Successfully tested.
- Any page or link rename:
  - [x] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [x] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [x] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [x] the *`Validate language on files added or modified`* step reports no vale warnings.

Change-Id: I05c188eea81bf9a9e58b1f74bc0ee37582c5b1da
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

**Comment by themr0c** [comment]
@benoitf For 3.0 you should target the `master` branch.

**Comment by benoitf** [comment]
@themr0c It's new content that is applying on what is on the current che version, ie = next version.
if you need to backport to other branch, it's up to documentation team.

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+IDE specific plug-ins are no longer part of devfile with devfile v2. Instead, it uses the specific meta-folder to describe the plug-ins.
+
+[id="visual-studio-code-extensions-json"]
+== Using `.vscode/extensions.json`
+
+The easiest way to add a Visual Studio Code extension is to add the extension to the `.vscode/extensions.json` file. Using Che-Theia the extension will be automatically installed and configured. With Che-Code, the Visual Studio Code fork, at the startup of the workspace, Visual Studio Code shows a recommendation pop-up to install the extension.
+
+.Procedure
+
+. Add `.vscode/extensions.json` file if missing at the root of the Git repository
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add `.vscode/extensions.json` file if it is missing at the root of the Git repository.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+. Add the extension using its identifier. Publisher and extension name are using a dot separated format.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add the extension using its identifier. Publisher and extension names use a dot-separated format.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+. Example adding the Visual Studio Code Java extension:
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Example of adding the Visual Studio Code Java extension:
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+```json
+{
+  "recommendations": [
+    "redhat.java"
+  ]
+}
+```
+
+. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
4. {prod-short} will automatically install the Red Hat Visual Studio Code Java extension in the {prod-short} workspace.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+
+Note: if the specified set of extensions IDs are not available in the {prod-short} registry, the workspace creation will not fail but the extensions will not be available.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
Note: If the specified set of extensions IDs is not available in the {prod-short} registry, the workspace creation will not fail but the extensions will not be available.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+Relying on `.vscode/extensions.json` is simple but sometimes it requires to specify some extra parameters for extensions.
+
+Here comes the optional `.che/che-theia-plugins.yaml` file definition
+
+=== Defines the extensions to install in the workspace
+
+.Procedure
+
+. Add `.che/che-theia-plugins.yaml` file if missing at the root of the Git repository
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add `.che/che-theia-plugins.yaml` file if it is missing at the root of the Git repository.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+. Add the id of the extension using the / separated format
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add the id of the extension using the '/' separated format.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+```YAML
+- id: redhat/java  
+```
+
+=== Change default memory limit
+
+For example, overriding some container settings, for example the memoryLimit for the RedHat Visual Studio Code Java extension.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
For example, override some container settings, like the memory limit for the RedHat Visual Studio Code Java extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+.Procedure
+
+. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc:65]:
```suggestion
1. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+. Specify the memory limit for the extension
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Specify the memory limit for the extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc (hunk 9 above)
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc:65]:
```suggestion
1. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
```

Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc (hunk 10 above)
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc:67]:
```suggestion
1. Specify the memory limit for the extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+
+```YAML
+- id: redhat/java  
+  override:  
+    sidecar:  
+      memoryLimit: 2Gi
+```
+
+. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace and increase the memory of the workspace by two gigabytes.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
3. {prod-short} will automatically install the Red Hat Visual Studio Code Java extension in the {prod-short} workspace and increase the memory of the workspace by two gigabytes.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+=== Override default preferences
+
+Override the default preferences of the Visual Studio Code extension for the workspace
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
Override the default preferences of the Visual Studio Code extension for the workspace.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+.Procedure
+
+. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+. Specify the preferences into `preferences` section
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Specify the preferences in the `Preferences` section.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+
+```YAML
+- id: redhat/java  
+  override:  
+    preferences:
+      java.server.launchMode: LightWeight
+```
+
+. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
3. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
```

File: modules/user-guide/pages/adding-a-vscode-extension.adoc [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+This definition is helpful when it is not possible to add extra files in the repository but is not the recommended way.
+
+Defines the Visual Studio Code extensions by inlining them in the devfile.
+
+=== Inlining `.vscode/extensions.json` file
+
+.Procedure
+
+. Edit `devfile.yaml` file and add if missing the `attributes` section at the root of the devfile
```
Comment (swatikarmarkar) [inline modules/user-guide/pages/adding-a-vscode-extension.adoc]:
```suggestion
1. Edit the `devfile.yaml` file and add if it is missing the `attributes` section at the root of the devfile.
```

**Comment by benoitf** [comment]
after applying changes, it doesn't pass anymore the checks
https://github.com/eclipse-che/che-docs/runs/5734309705?check_suite_focus=true

**Comment by benoitf** [comment]
Pull request against master branch has been moved to https://github.com/eclipse-che/che-docs/pull/2265

**Pull request closed without merging**